library/md.c

```diff
@@ -200,12 +206,11 @@ void mbedtls_md_free( mbedtls_md_context_t *ctx )
 int mbedtls_md_clone( mbedtls_md_context_t *dst,
                       const mbedtls_md_context_t *src )
 {
-    if( dst == NULL || dst->md_info == NULL ||
-        src == NULL || src->md_info == NULL ||
-        dst->md_info != src->md_info )
-    {
-        return( MBEDTLS_ERR_MD_BAD_INPUT_DATA );
-    }
+    MBEDTLS_MD_VALIDATE_RET( dst != NULL );
+    MBEDTLS_MD_VALIDATE_RET( src != NULL );
+    MBEDTLS_MD_VALIDATE_RET( dst->md_info != NULL );
```

Comment:
Non-blocking: In line with the other PRs, I'd prefer to go with the first two non-NULL checks only.